consumer/consumererror/combine.go

```diff
@@ -15,39 +15,76 @@
 package consumererror
 
 import (
-	"fmt"
+	"errors"
 	"strings"
 )
 
+type combined []error
+
+var (
+	_ error                                    = (*combined)(nil)
+	_ interface{ Is(target error) bool }       = (*combined)(nil)
+	_ interface{ As(target interface{}) bool } = (*combined)(nil)
+)
+
+func (c combined) Error() string {
+	var sb strings.Builder
+	length := len(c)
+	for i, err := range c {
+		sb.WriteString(err.Error())
+		if i != length-1 {
+			sb.WriteString("; ")
+		}
+	}
+	return "[" + sb.String() + "]"
+}
+
+// Range iterates over the slice and all subslices of errors
+// and will apply the passed function on each.
+// The method will return once a the passed function returns true
+// or there is no items in the slice.
+func (c combined) Range(fn func(error) bool) bool {
```

Comment:
Why do you need this public method? Can it be private?

Reply:
I think for sure it can since the struct is also private so this will not be exposed.

Reply:
No reason for it to be public, I just need to rename to make it private since it is a keyword.